lib/Fmt_ast.ml

```diff
@@ -2328,12 +2328,12 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
            $ fmt_atrs ))
   | Pexp_coerce (e1, t1, t2) ->
       hvbox 2
-        (wrap_fits_breaks ~space:false c.conf "(" ")"
-           ( fmt_expression c (sub_exp ~ctx e1)
-           $ opt t1 (fmt "@ : " >$ (sub_typ ~ctx >> fmt_core_type c))
-           $ fmt "@ :> "
-           $ fmt_core_type c (sub_typ ~ctx t2)
-           $ fmt_atrs ))
+        ( wrap_fits_breaks ~space:false c.conf "(" ")"
+            ( fmt_expression c (sub_exp ~ctx e1)
+            $ opt t1 (fmt "@ : " >$ (sub_typ ~ctx >> fmt_core_type c))
+            $ fmt "@ :> "
+            $ fmt_core_type c (sub_typ ~ctx t2) )
+        $ fmt_atrs )
```

Comment:
The whole expression must be wrapped in parentheses if there are attributes (with `wrap_if parens`).

Example:
```ocaml
f ((x :> t) [@a])
```

Reply:
Ah, I see. I added a failing test for this. I don't see how to fix this, though. `wrap_if parens` adds parens in other cases, and using `wrap_fits_break_if` does not seem to do the trick.

Reply:
What we need to do in this case is wrap again in parentheses. The `wrap_fits_breaks` stays here, we need an other wrap around `wrap_fits_breaks .. $ fmt_atrs`.

Is it adding parentheses when not needed ? In that case, we should use `wrap_if (parens && has_attr)`. In the cases parentheses are required and there is no attribute (`parens = true`, `has_attr = false`), the inner parentheses are enough.

Reply:
`parens && has_attr` did the trick, thanks!